keras_nlp/src/models/stable_diffusion_3/stable_diffusion_3_text_to_image.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from keras import ops
+from keras import random
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.clip.clip_preprocessor import CLIPPreprocessor
+from keras_nlp.src.models.stable_diffusion_3.stable_diffusion_3_backbone import (
+    StableDiffusion3Backbone,
+)
+from keras_nlp.src.models.t5.t5_preprocessor import T5Preprocessor
+from keras_nlp.src.models.text_to_image import TextToImage
+
+
+@keras_nlp_export("keras_nlp.models.StableDiffusion3TextToImage")
+class StableDiffusion3TextToImage(TextToImage):
+    backbone_cls = StableDiffusion3Backbone
+    clip_l_preprocessor_cls = CLIPPreprocessor
```

Comment:
I think instead of this design we should keep our preprocessing and task objects one to one. So one `StableDiffusion3TextToImagePreprocessor` that stitches together all these inputs. We might need to add some extra bits in the subclass to support multiple tokenizers, but that should be doable. We want to keep `task.preprocessor` as a individual layer that does everything we need here.

Reply:
I have added `StableDiffusion3TextToImagePreprocessor` and it works well.